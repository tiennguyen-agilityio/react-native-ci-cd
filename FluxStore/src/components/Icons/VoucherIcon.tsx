import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const VoucherIcon = memo(
  ({width = 19, height = 15, color, disabled = false, style, onPress}: IconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="voucher-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 19 15" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M18.064 6.222a.643.643 0 01-.449.18c-.626 0-1.134.494-1.134 1.094 0 .604.502 1.095 1.122 1.101.35.004.647.24.647.58v2.109c0 1.774-1.48 3.214-3.309 3.214h-2.759a.509.509 0 01-.516-.502v-1.775a.62.62 0 00-.634-.617.625.625 0 00-.635.617v1.775a.509.509 0 01-.515.502H4.059C2.239 14.5.75 13.061.75 11.286V9.177c0-.34.297-.576.647-.58.62-.006 1.122-.497 1.122-1.101 0-.584-.491-1.028-1.134-1.028a.644.644 0 01-.449-.18.607.607 0 01-.186-.436v-2.13C.75 1.95 2.243.5 4.067.5h5.815c.284 0 .515.224.515.501v2.105a.63.63 0 00.635.617c.355 0 .634-.28.634-.617V1.001c0-.277.231-.501.516-.501h2.76c1.827 0 3.308 1.439 3.308 3.214v2.072c0 .164-.068.32-.186.436zm-7.032 3.79c.355 0 .634-.28.634-.617V6.107a.625.625 0 00-.634-.617.63.63 0 00-.635.617v3.288a.63.63 0 00.635.617z"
            fill={color || icon.default}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
