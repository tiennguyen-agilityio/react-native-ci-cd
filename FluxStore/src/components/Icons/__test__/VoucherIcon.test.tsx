import {render, fireEvent, screen} from '@testing-library/react-native';

import {VoucherIcon} from '../VoucherIcon';

describe('VoucherIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<VoucherIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive true', () => {
    const {toJSON} = render(<VoucherIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<VoucherIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('voucher-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<VoucherIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('voucher-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
