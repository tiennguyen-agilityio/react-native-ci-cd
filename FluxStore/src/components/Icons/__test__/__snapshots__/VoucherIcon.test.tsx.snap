// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoucherIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="voucher-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={19}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 19,
        },
      ]
    }
    vbHeight={15}
    vbWidth={19}
    width={19}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M18.064 6.222a.643.643 0 01-.449.18c-.626 0-1.134.494-1.134 1.094 0 .604.502 1.095 1.122 1.101.35.004.647.24.647.58v2.109c0 1.774-1.48 3.214-3.309 3.214h-2.759a.509.509 0 01-.516-.502v-1.775a.62.62 0 00-.634-.617.625.625 0 00-.635.617v1.775a.509.509 0 01-.515.502H4.059C2.239 14.5.75 13.061.75 11.286V9.177c0-.34.297-.576.647-.58.62-.006 1.122-.497 1.122-1.101 0-.584-.491-1.028-1.134-1.028a.644.644 0 01-.449-.18.607.607 0 01-.186-.436v-2.13C.75 1.95 2.243.5 4.067.5h5.815c.284 0 .515.224.515.501v2.105a.63.63 0 00.635.617c.355 0 .634-.28.634-.617V1.001c0-.277.231-.501.516-.501h2.76c1.827 0 3.308 1.439 3.308 3.214v2.072c0 .164-.068.32-.186.436zm-7.032 3.79c.355 0 .634-.28.634-.617V6.107a.625.625 0 00-.634-.617.63.63 0 00-.635.617v3.288a.63.63 0 00.635.617z"
        fill={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`VoucherIcon renders correctly (matches snapshot) with isActive true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="voucher-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={19}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 19,
        },
      ]
    }
    vbHeight={15}
    vbWidth={19}
    width={19}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M18.064 6.222a.643.643 0 01-.449.18c-.626 0-1.134.494-1.134 1.094 0 .604.502 1.095 1.122 1.101.35.004.647.24.647.58v2.109c0 1.774-1.48 3.214-3.309 3.214h-2.759a.509.509 0 01-.516-.502v-1.775a.62.62 0 00-.634-.617.625.625 0 00-.635.617v1.775a.509.509 0 01-.515.502H4.059C2.239 14.5.75 13.061.75 11.286V9.177c0-.34.297-.576.647-.58.62-.006 1.122-.497 1.122-1.101 0-.584-.491-1.028-1.134-1.028a.644.644 0 01-.449-.18.607.607 0 01-.186-.436v-2.13C.75 1.95 2.243.5 4.067.5h5.815c.284 0 .515.224.515.501v2.105a.63.63 0 00.635.617c.355 0 .634-.28.634-.617V1.001c0-.277.231-.501.516-.501h2.76c1.827 0 3.308 1.439 3.308 3.214v2.072c0 .164-.068.32-.186.436zm-7.032 3.79c.355 0 .634-.28.634-.617V6.107a.625.625 0 00-.634-.617.63.63 0 00-.635.617v3.288a.63.63 0 00.635.617z"
        fill={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
